xds/src/main/java/io/grpc/xds/RingHashLoadBalancer.java

```diff
@@ -207,6 +219,28 @@ private boolean validateAddrList(List<EquivalentAddressGroup> addrList) {
     return true;
   }
 
+  @Nullable
+  private String validateNoDuplicateAddresses(List<EquivalentAddressGroup> addrList) {
+    Set<SocketAddress> addresses = new HashSet<>();
+    Multiset<String> dups = HashMultiset.create();
+    for (EquivalentAddressGroup eag : addrList) {
+      for (SocketAddress address : eag.getAddresses()) {
+        if (!addresses.add(address)) {
+          dups.add(address.toString());
+        }
+      }
+    }
+
+    if (!dups.isEmpty()) {
+      return dups.entrySet().stream()
+          .map((entry) ->
+              String.format("Address: %s, count: %d", entry.getElement(), entry.getCount() + 1))
```

Comment:
totally a nit - but I think this makes it more self-evident why we add one to the count.
```suggestion
          .map((dup) ->
              String.format("Address: %s, count: %d", dup.getElement(), dup.getCount() + 1))
```

Reply:
Good point